runtime/gc_realtime/RealtimeRootScanner.hpp

```diff
@@ -117,5 +118,377 @@ class MM_RealtimeRootScanner : public MM_RootScanner
 	}
 };
 
+/**
+ * This scanner will mark objects that pass through its doSlot.
+ */
+class MM_RealtimeMarkingSchemeRootMarker : public MM_RealtimeRootScanner
+{
+/* Data memebers / types */
+public:
+protected:
+private:
+	
+/* Methods */
+public:
+
+	/**
+	 * Simple chained constructor.
+	 */
+	MM_RealtimeMarkingSchemeRootMarker(MM_EnvironmentRealtime *env, MM_RealtimeGC *realtimeGC) :
+		MM_RealtimeRootScanner(env, realtimeGC)
+	{
+		_typeId = __FUNCTION__;
+	}
+	
+	/**
+	 * This scanner can be instantiated so we must give it a name.
+	 */
+	virtual const char*
+	scannerName()
+	{
+		return "Mark";
+	}
+	
+	/**
+	 * Wraps the scanning of one thread to only happen if it hasn't already occured in this phase of this GC,
+	 * also sets the thread up for the upcoming forwarding phase.
+	 * @return true if the thread was scanned (the caller should offer to yield), false otherwise.
+	 * @see MM_RootScanner::scanOneThread()
+	 */
+	virtual void
+	scanOneThreadImpl(MM_EnvironmentRealtime *env, J9VMThread* walkThread, void* localData)
+	{
+		MM_EnvironmentRealtime* walkThreadEnv = MM_EnvironmentRealtime::getEnvironment(walkThread->omrVMThread);
+		MM_RealtimeGC* realtimeGC = (MM_RealtimeGC*)_realtimeGC;
```

Comment:
The cast is not necessary because `_realtimeGC` has type `(MM_RealtimeGC*)` already. I would not declare `realtimeGC` at all but use `_realtimeGC` instead.